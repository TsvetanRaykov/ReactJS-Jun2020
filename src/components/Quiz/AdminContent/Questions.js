import React from 'react'
import {
	List,
	ListItem,
	ListItemText,
	ListItemSecondaryAction,
	IconButton,
	Tooltip,
} from '@material-ui/core'
import { Edit } from '@material-ui/icons'

const Questions = (props) => {
	const { questions, editQuestion, activeQuestionIndex } = props

	return (
		<List>
			{questions.map(({ question }, i) => {
				const styles = i === activeQuestionIndex ? { display: 'none' } : {}
				return (
					<div style={styles} key={question}>
						<ListItem>
							<ListItemText primary={question} />
							<ListItemSecondaryAction>
								<Tooltip title='Edit question and answers' placement='left'>
									<IconButton
										edge='end'
										aria-label='edit'
										color='primary'
										onClick={() => editQuestion(i)}
									>
										<Edit />
									</IconButton>
								</Tooltip>
							</ListItemSecondaryAction>
						</ListItem>
					</div>
				)
			})}
		</List>
	)
}

export default Questions
