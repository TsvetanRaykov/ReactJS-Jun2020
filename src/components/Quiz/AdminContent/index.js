import React, { useContext, useState, useEffect } from 'react'
import { Paper, Typography, withStyles } from '@material-ui/core'
import SetQuestionForm from '../../Forms/Quiz/SetQuestion'
import Context from '../../../Context'
import Questions from './Questions'
import CustomPaginationActionsTable from './QuestionsTable'

const styles = (theme) => ({
	root: {
		marginTop: theme.spacing(2),
	},
})

const AdminContent = (props) => {
	const { classes, formOpen, formHandler, newQuestionType } = props

	const {
		quiz: { questions },
	} = useContext(Context)

	const [activeQuestionIndex, setActiveQuestion] = useState(-1)

	const [questionList, setQuestionList] = useState(questions.slice(0))

	const formClose = () => {
		setQuestionList(questions)
		formHandler(false)
		setActiveQuestion(-1)
	}

	const editQuestion = (i) => {
		setActiveQuestion(i)
		formHandler(true)
	}

	useEffect(() => {
		setQuestionList(questions)
	}, [questions])

	return (
		<>
			<Paper className={classes.root}>
				<Typography component='h1' variant='h6'>
					Questions
				</Typography>
				{/* <Questions
					questions={questionList}
					editQuestion={editQuestion}
					activeQuestionIndex={activeQuestionIndex}
				/> */}
				<CustomPaginationActionsTable
					questions={questionList}
					editQuestion={editQuestion}
					activeQuestionIndex={activeQuestionIndex}
				/>
			</Paper>
			{formOpen && (
				<SetQuestionForm
					formClose={formClose}
					activeQuestionIndex={activeQuestionIndex}
					newQuestionType={newQuestionType}
				/>
			)}
		</>
	)
}

export default withStyles(styles)(AdminContent)
