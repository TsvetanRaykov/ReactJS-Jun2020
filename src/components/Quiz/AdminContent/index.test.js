import React from 'react'
import renderer from 'react-test-renderer'
import TestWrapper from '../../../utils/test-wrapper'
import AdminContent from '.'
import Questions from './Questions'

jest.mock('react', () => ({
	...jest.requireActual('react'),
	useLayoutEffect: jest.requireActual('react').useEffect,
}))

describe('<AdminContent />', () => {
	it('Should match the snapshot', () => {
		const wrapper = renderer.create(
			<TestWrapper>
				<AdminContent />
			</TestWrapper>
		)
		expect(wrapper).toMatchSnapshot()
	})
})

describe('<Questions />', () => {
	it('Should match the snapshot', () => {
		const wrapper = renderer.create(
			<TestWrapper>
				<Questions questions={[]} />
			</TestWrapper>
		)
		expect(wrapper).toMatchSnapshot()
	})
})
