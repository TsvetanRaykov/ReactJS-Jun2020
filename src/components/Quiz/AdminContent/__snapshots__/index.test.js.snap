// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AdminContent /> Should match the snapshot 1`] = `
<div
  className="MuiPaper-root AdminContent-root-1 MuiPaper-elevation1 MuiPaper-rounded"
>
  <h1
    className="MuiTypography-root MuiTypography-h6"
  >
    Questions
  </h1>
  <ul
    className="MuiList-root MuiList-padding"
  />
</div>
`;

exports[`<Questions /> Should match the snapshot 1`] = `
<ul
  className="MuiList-root MuiList-padding"
/>
`;
